src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -651,24 +651,33 @@ private static ArraySegment<char> Base64UrlEncode(ReadOnlySpan<byte> input)
 
             if (failsPolicyChecks)
             {
-                X509ChainElement leafElement = elements[0];
+                if (overallStatus == null)
+                {
+                    overallStatus = new List<X509ChainStatus>();
+                }
 
                 X509ChainStatus chainStatus = new X509ChainStatus
                 {
                     Status = X509ChainStatusFlags.NotValidForUsage,
                     StatusInformation = SR.Chain_NoPolicyMatch,
                 };
 
-                var elementStatus = new List<X509ChainStatus>(leafElement.ChainElementStatus.Length + 1);
-                elementStatus.AddRange(leafElement.ChainElementStatus);
-
-                AddUniqueStatus(elementStatus, ref chainStatus);
                 AddUniqueStatus(overallStatus, ref chainStatus);
 
-                elements[0] = new X509ChainElement(
-                    leafElement.Certificate,
-                    elementStatus.ToArray(),
-                    leafElement.Information);
+                for (int i = 0; i < elements.Length; i++)
+                {
+                    X509ChainElement element = elements[i];
+
+                    var elementStatus = new List<X509ChainStatus>(element.ChainElementStatus.Length + 1);
```

Comment:
Can we instead move `elementStatus` to be declared outside of the loop, and then make this:
```C#
if (elementStatus == null)
{
    elementStatus = new List<X509ChainStatus>(element.ChainElementStatus.Length + 1);
}
else
{
    elementStatus.Clear();
}
```
or something like that?